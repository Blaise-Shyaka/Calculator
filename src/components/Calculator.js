import { useState } from 'react';
import Display from './Display';
import ButtonPanel from './ButtonPanel';
import calculate from '../logic/calculate';
import * as styles from '../styles/app.module.css';

const Calculator = () => {
  const [calculatorState, setCalculatorState] = useState({
    total: null,
    next: null,
    operation: null,
  });

  function handleClick(buttonName) {
    const result = calculate(calculatorState, buttonName);
    setCalculatorState(result);
  }

  const { total, next } = calculatorState;
  const result = next ?? total;
  return (
    <div className={styles.app}>
      <Display result={result ?? Number(result).toString()} />
      <ButtonPanel clickHandler={handleClick} />
    </div>
  );
};

export default Calculator;
