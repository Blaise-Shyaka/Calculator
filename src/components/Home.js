import magician from '../assets/magician.png';
import * as homeStyles from '../styles/home.module.css';

export default function Home() {
  return (
    <div className={homeStyles.home}>
      <div>
        <header>
          <h1>Where Math Magic Happens</h1>
        </header>
        <main>
          <p>
            Lorem ipsum dolor sit amet, consectetur adipisicing elit,
            sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.
            Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris
            nisi ut aliquip ex ea commodo consequat.
            Duis aute irure dolor in reprehenderit in voluptate velit
            esse cillum dolore eu fugiat nulla pariatur.
            Excepteur sint occaecat cupidatat non proident,
            sunt in culpa qui o!cia deserunt mollit anim id est laborum. Ipsu
          </p>
          <p>
            Lorem ipsum dolor sit amet, consectetur adipisicing elit,
            sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut
            enim ad minim veniam, quis nostrud exercitation ullamco laboris
            nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in
            reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur.
            Excepteur sint occaecat cupidatat non proident, sunt in
            culpa qui o!cia deserunt mollit anim id est laborum. Ipsu
          </p>
        </main>
      </div>
      <aside>
        <img src={magician} alt="Magician Dealing with Numbers" />
      </aside>
    </div>
  );
}
