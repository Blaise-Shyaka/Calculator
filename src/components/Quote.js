export default function Quote() {
  return (
    <div className="align">
      <h1>Interesting Maths Quotes</h1>
      <blockquote className="italics">
        Mathematics is the most beautiful and most powerful creation of the human spirit.
      </blockquote>
      <blockquote className="italics">
        Mathematics is the art of giving the same name to different things.
      </blockquote>
      <blockquote className="italics">
        The beauty of mathematics only shows itself to more patient followers.
      </blockquote>
      <blockquote className="italics">
        To solve math problems,
        you need to know the basic mathematics before you can start applying it.
      </blockquote>
      <blockquote className="italics">
        Mathematics knows no races or geographic boundaries;
        for mathematics, the cultural world is one country.
      </blockquote>
      <blockquote className="italics">
        Obvious is the most dangerous word in mathematics.
      </blockquote>
    </div>
  );
}
