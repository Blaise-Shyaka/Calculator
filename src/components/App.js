import { Route, Switch } from 'react-router-dom';
import Home from './Home';
import Quote from './Quote';
import Calculator from './Calculator';
import Navigation from './Navigation';

export default function App() {
  return (
    <div>
      <Navigation />
      <Switch>
        <Route exact path="/" component={Home} />
        <Route exact path="/calculator" component={Calculator} />
        <Route exact path="/quotes" component={Quote} />
      </Switch>
    </div>
  );
}
