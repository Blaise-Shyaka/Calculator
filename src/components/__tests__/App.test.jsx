import { render, fireEvent } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import TestRenderer from 'react-test-renderer';
import App from '../App';
import Home from '../Home';
import Calculator from '../Calculator';
import Quote from '../Quote';

describe('App Component', () => {
  const appComponent = (
    <BrowserRouter>
      <App />
    </BrowserRouter>
  );
  it('Should render the App component successfully', () => {
    render(appComponent);
  });

  it('The Home page should be rendered by default', () => {
    const { getByText } = render(appComponent);
    getByText('Where Math Magic Happens');
  });

  it('Should display the navbar', () => {
    const { getByText } = render(appComponent);
    getByText('Home');
    getByText('Calculator');
    getByText('Quotes');
  });

  it('Should navigate to the calculator page if "Calculator" tab is clicked', () => {
    const { getByText, getAllByRole } = render(appComponent);
    const btn = getByText('Calculator');
    fireEvent.click(btn);
    const calculatorButtons = getAllByRole('button');
    expect(calculatorButtons.length).toBe(19);
  });

  it('Should navigate to the quotes page if "Quotes" tab is clicked', () => {
    const { getByText } = render(appComponent);
    const btn = getByText('Quotes');
    fireEvent.click(btn);
    getByText('Interesting Maths Quotes');
  });

  it('Should render the Home page correctly', () => {
    const tree = TestRenderer.create(<Home />);
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('Should render the Calculator page correctly', () => {
    const tree = TestRenderer.create(<Calculator />);
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('Should render the Quotes page correctly', () => {
    const tree = TestRenderer.create(<Quote />);
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
