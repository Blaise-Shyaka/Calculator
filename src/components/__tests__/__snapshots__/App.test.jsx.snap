// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component Should render the Calculator page correctly 1`] = `
<div>
  <div>
    <p>
      0
    </p>
  </div>
  <div>
    <div>
      <button
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "",
            "border": "1px solid #ccc",
            "fontWeight": 600,
            "height": "100%",
            "outline": "none",
            "width": "25%",
          }
        }
        type="button"
      >
        AC
      </button>
      <button
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "",
            "border": "1px solid #ccc",
            "fontWeight": 600,
            "height": "100%",
            "outline": "none",
            "width": "25%",
          }
        }
        type="button"
      >
        +/-
      </button>
      <button
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "",
            "border": "1px solid #ccc",
            "fontWeight": 600,
            "height": "100%",
            "outline": "none",
            "width": "25%",
          }
        }
        type="button"
      >
        %
      </button>
      <button
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "orange",
            "border": "1px solid #ccc",
            "fontWeight": 600,
            "height": "100%",
            "outline": "none",
            "width": "25%",
          }
        }
        type="button"
      >
        รท
      </button>
    </div>
    <div>
      <button
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "",
            "border": "1px solid #ccc",
            "fontWeight": 600,
            "height": "100%",
            "outline": "none",
            "width": "25%",
          }
        }
        type="button"
      >
        7
      </button>
      <button
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "",
            "border": "1px solid #ccc",
            "fontWeight": 600,
            "height": "100%",
            "outline": "none",
            "width": "25%",
          }
        }
        type="button"
      >
        8
      </button>
      <button
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "",
            "border": "1px solid #ccc",
            "fontWeight": 600,
            "height": "100%",
            "outline": "none",
            "width": "25%",
          }
        }
        type="button"
      >
        9
      </button>
      <button
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "orange",
            "border": "1px solid #ccc",
            "fontWeight": 600,
            "height": "100%",
            "outline": "none",
            "width": "25%",
          }
        }
        type="button"
      >
        X
      </button>
    </div>
    <div>
      <button
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "",
            "border": "1px solid #ccc",
            "fontWeight": 600,
            "height": "100%",
            "outline": "none",
            "width": "25%",
          }
        }
        type="button"
      >
        4
      </button>
      <button
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "",
            "border": "1px solid #ccc",
            "fontWeight": 600,
            "height": "100%",
            "outline": "none",
            "width": "25%",
          }
        }
        type="button"
      >
        5
      </button>
      <button
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "",
            "border": "1px solid #ccc",
            "fontWeight": 600,
            "height": "100%",
            "outline": "none",
            "width": "25%",
          }
        }
        type="button"
      >
        6
      </button>
      <button
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "orange",
            "border": "1px solid #ccc",
            "fontWeight": 600,
            "height": "100%",
            "outline": "none",
            "width": "25%",
          }
        }
        type="button"
      >
        -
      </button>
    </div>
    <div>
      <button
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "",
            "border": "1px solid #ccc",
            "fontWeight": 600,
            "height": "100%",
            "outline": "none",
            "width": "25%",
          }
        }
        type="button"
      >
        1
      </button>
      <button
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "",
            "border": "1px solid #ccc",
            "fontWeight": 600,
            "height": "100%",
            "outline": "none",
            "width": "25%",
          }
        }
        type="button"
      >
        2
      </button>
      <button
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "",
            "border": "1px solid #ccc",
            "fontWeight": 600,
            "height": "100%",
            "outline": "none",
            "width": "25%",
          }
        }
        type="button"
      >
        3
      </button>
      <button
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "orange",
            "border": "1px solid #ccc",
            "fontWeight": 600,
            "height": "100%",
            "outline": "none",
            "width": "25%",
          }
        }
        type="button"
      >
        +
      </button>
    </div>
    <div>
      <button
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "",
            "border": "1px solid #ccc",
            "fontWeight": 600,
            "height": "100%",
            "outline": "none",
            "width": "50%",
          }
        }
        type="button"
      >
        0
      </button>
      <button
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "",
            "border": "1px solid #ccc",
            "fontWeight": 600,
            "height": "100%",
            "outline": "none",
            "width": "25%",
          }
        }
        type="button"
      >
        .
      </button>
      <button
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "orange",
            "border": "1px solid #ccc",
            "fontWeight": 600,
            "height": "100%",
            "outline": "none",
            "width": "25%",
          }
        }
        type="button"
      >
        =
      </button>
    </div>
  </div>
</div>
`;

exports[`App Component Should render the Home page correctly 1`] = `
<div>
  <div>
    <header>
      <h1>
        Where Math Magic Happens
      </h1>
    </header>
    <main>
      <p>
        Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui o!cia deserunt mollit anim id est laborum. Ipsu
      </p>
      <p>
        Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui o!cia deserunt mollit anim id est laborum. Ipsu
      </p>
    </main>
  </div>
  <aside>
    <img
      alt="Magician Dealing with Numbers"
      src="magician.png"
    />
  </aside>
</div>
`;

exports[`App Component Should render the Quotes page correctly 1`] = `
<div
  className="align"
>
  <h1>
    Interesting Maths Quotes
  </h1>
  <blockquote
    className="italics"
  >
    Mathematics is the most beautiful and most powerful creation of the human spirit.
  </blockquote>
  <blockquote
    className="italics"
  >
    Mathematics is the art of giving the same name to different things.
  </blockquote>
  <blockquote
    className="italics"
  >
    The beauty of mathematics only shows itself to more patient followers.
  </blockquote>
  <blockquote
    className="italics"
  >
    To solve math problems, you need to know the basic mathematics before you can start applying it.
  </blockquote>
  <blockquote
    className="italics"
  >
    Mathematics knows no races or geographic boundaries; for mathematics, the cultural world is one country.
  </blockquote>
  <blockquote
    className="italics"
  >
    Obvious is the most dangerous word in mathematics.
  </blockquote>
</div>
`;
